File: src/BukkitPE/level/generator/GeneratorRegisterTask.php
<?php
namespace BukkitPE\level\generator;

use BukkitPE\block\Block;

use BukkitPE\level\generator\biome\Biome;
use BukkitPE\level\Level;
use BukkitPE\level\SimpleChunkManager;
use BukkitPE\scheduler\AsyncTask;

use BukkitPE\utils\Random;

class GeneratorRegisterTask extends AsyncTask{

	public $generator;
	public $settings;
	public $seed;
	public $levelId;

	public function __construct(Level $level, Generator $generator){
		$this->generator = get_class($generator);
		$this->settings = $generator->getSettings();
		$this->seed = $level->getSeed();
		$this->levelId = $level->getId();
	}

	public function onRun(){
		Block::init();
		Biome::init();
		$manager = new SimpleChunkManager($this->seed);
		$this->saveToThreadStore("generation.level{$this->levelId}.manager", $manager);
		/** @var Generator $generator */
		$generator = $this->generator;
		$generator = new $generator($this->settings);
		$generator->init($manager, new Random($manager->getSeed()));
		$this->saveToThreadStore("generation.level{$this->levelId}.generator", $generator);
	}
}

File: src/main/java/net/BukkitPE/level/generator/task/GeneratorRegisterTask.java
package net.BukkitPE.level.generator.task;

import net.BukkitPE.block.Block;
import net.BukkitPE.level.Level;
import net.BukkitPE.level.SimpleChunkManager;
import net.BukkitPE.level.generator.Generator;
import net.BukkitPE.level.generator.biome.Biome;
import net.BukkitPE.math.BukkitPERandom;
import net.BukkitPE.scheduler.AsyncTask;

import java.util.Map;

/**
 * BukkitPE Project
 */
public class GeneratorRegisterTask extends AsyncTask {

    public final Class<? extends Generator> generator;
    public final Map<String, Object> settings;
    public final long seed;
    public final int levelId;

    public GeneratorRegisterTask(Level level, Generator generator) {
        this.generator = generator.getClass();
        this.settings = generator.getSettings();
        this.seed = level.getSeed();
        this.levelId = level.getId();
    }

    @Override
    public void onRun() {
        Block.init();
        Biome.init();
        SimpleChunkManager manager = new SimpleChunkManager(this.seed);
        try {
            Generator generator = this.generator.getConstructor(Map.class).newInstance(this.settings);
            generator.init(manager, new BukkitPERandom(manager.getSeed()));
            GeneratorPool.put(this.levelId, generator);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}
